src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -280,6 +289,64 @@ public void shutdown(ExecutorDriver d) {
     log.info("Executor asked to shutdown");
   }
 
+  protected void runHealthChecks(ExecutorDriver driver, Task task) {
+    log.info("Performing health check for task: " + task.taskInfo.getTaskId().getValue());
+    boolean taskHealthy = false;
+    String taskIdStr = task.taskInfo.getTaskId().getValue();
+    int healthCheckPort = -1;
+
+    if (taskIdStr.contains(HDFSConstants.DATA_NODE_ID)) {
+      healthCheckPort = HDFSConstants.DATA_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.JOURNAL_NODE_ID)) {
+      healthCheckPort = HDFSConstants.JOURNAL_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.ZKFC_NODE_ID)) {
+      healthCheckPort = HDFSConstants.ZKFC_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.NAME_NODE_ID)) {
+      healthCheckPort = HDFSConstants.NAME_NODE_PORT;
+    } else {
+      log.error("Task unknown: " + taskIdStr);
+    }
+    if (healthCheckPort != -1) {
+      Socket socket = null;
+      try
+      {
+        String hostAddress = InetAddress.getLocalHost().getHostAddress();
+        socket = new Socket();
+        socket.bind(new InetSocketAddress(hostAddress, healthCheckPort));
```

Comment:
Testing with a bind instead of actually hitting that host:port? What if the task is hung but still bound to the port? This seems like a very poor health check, no better than a 'ps aux |grep datanode'.. Can we do better, at least in a subsequent PR?